src/MISC/fix_eco_force.cpp

```diff
@@ -0,0 +1,642 @@
+  if (me == 0) {
+    char line[IMGMAX];
+    char *result;
+    int count;
+
+    FILE *infile = fopen(dir_filename, "r");
+    if (infile == NULL) error->one(FLERR, "Fix eco/force file open failed");
+    for (int i = 0; i < 6; ++i) {
+      result = fgets(line, IMGMAX, infile);
+      if (!result) error->one(FLERR, "Fix eco/force file read failed");
+      count = sscanf(line, "%lg %lg %lg", &dir_vec[i][0], &dir_vec[i][1], &dir_vec[i][2]);
+      if (count != 3) error->one(FLERR, "Fix eco/force file read failed");
+    }
+    fclose(infile);
+
+    // calculate reciprocal lattice vectors
+    get_reciprocal();
+
+    squared_cutoff = r_cut * r_cut;
+    inv_squared_cutoff = 1.0 / squared_cutoff;
+    half_u = 0.5 * u_0;
+    inv_eta = 1.0 / eta;
+  }
+
+  // initializations
+  MPI_Bcast(&dir_vec[0][0], 18, MPI_DOUBLE, 0, world);                  // communicate direct lattice vectors
+  MPI_Bcast(&reciprocal_vectors[0][0][0], 18, MPI_DOUBLE, 0, world);    // communicate reciprocal vectors
+  MPI_Bcast(&squared_cutoff, 1, MPI_DOUBLE, 0, world);                  // communicate squared cutoff radius
+  MPI_Bcast(&inv_squared_cutoff, 1, MPI_DOUBLE, 0, world);              // communicate inverse squared cutoff radius
+  MPI_Bcast(&half_u, 1, MPI_DOUBLE, 0, world);                          // communicate half potential energy
+  MPI_Bcast(&inv_eta, 1, MPI_DOUBLE, 0, world);                         // communicate inverse threshold
+
+  // set comm size needed by this Fix
+  if (u_0 != 0.0) comm_forward = 110;
+  else comm_forward = 98;
+
+  added_energy = 0.0;                                   // initial energy
+
+  nmax = atom->nmax;
+  nbr = (Nbr *) memory->smalloc(nmax * sizeof(Nbr), "eco/force:nbr");
+  memory->create(order, nmax, 2, "eco/force:order");
+  array_atom = order;
+
+  // zero the array since a variable may access it before first run
+  for (int i = 0; i < atom->nlocal; ++i) order[i][0] = order[i][1] = 0.0;
+}
+
+/* ---------------------------------------------------------------------- */
+
+FixECOForce::~FixECOForce() {
+  memory->destroy(order);
+  memory->sfree(nbr);
+  delete[] dir_filename;
+}
+
+/* ---------------------------------------------------------------------- */
+
+int FixECOForce::setmask() {
+  int mask = 0;
+  mask |= POST_FORCE;
+  mask |= THERMO_ENERGY;
+  mask |= POST_FORCE_RESPA;
+  return mask;
+}
+
+/* ---------------------------------------------------------------------- */
+
+void FixECOForce::init() {
+  // get this processors rank
+  MPI_Comm_rank(world, &me);
+
+  // compute normalization factor
+  if (me == 0) {
+    int neigh;
+    get_norm(neigh);
+    printf("fix eco/force: cutoff=%f norm_fac=%f neighbors=%i\n", r_cut, norm_fac, neigh);
+  }
+  
+  inv_norm_fac = 1.0 / norm_fac;
+
+  // check parameters
+  if (r_cut > force->pair->cutforce) {
+    error->all(FLERR, "Pair cutoff too small: use 'hybrid/overlay' to set the pair cutoff larger than the fix eco/force cutoff");
```

Comment:
why do you need to have an increased force cutoff? and why through this unusual method?